style: apply /style-guide pass to weave (#2739)
wandb/docs: weave/guides/tools/weave-in-workspaces.mdx

---
title: "Use Weave with W&B training runs"
description: "Integrate Weave traces with W&B training runs to view function execution details alongside ML metrics in workspace dashboards."
keywords: ["workspaces", "training runs", "traces", "wandb.run", "artifacts"]
---

# Log traces during model training runs

> 

Reviewer: ```suggestion
You can log Weave traces during model training runs and view them in your personal workspace.
```

Author: Applied in 993de5e91 — moved this intro sentence above the screenshot.

<Frame>
![A W&B workspace showing a Weave trace panel alongside training run metrics.](/images/weave/traces-in-workspace.png)
</Frame>

> You can log Weave traces during model training runs and view them in your personal workspace.

Reviewer: ```suggestion
```

Author: Applied in 993de5e91 — removed the duplicate sentence after the Frame (it now sits above the screenshot).

To add an artifact as an attribute of a `Model`, pass the artifact URL when you instantiate the model:

<Note>

> The Weave Python SDK is the only SDK that supports the `Model` class.

Reviewer: ```suggestion
The `Model` class is only supported in the Weave Python SDK.
```

Author: Applied in 6cbc089f8 (your committed suggestion).
